as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/UnitedServerListenerManager.java

```diff
@@ -71,32 +80,42 @@ public static boolean isJBossServer(IServer server) {
 	}
 
 
-	public synchronized void addListener(UnitedServerListener listener) {
-		if( !list.contains(listener)) {
-			list.add(listener);
-			// united listener manager is already initialized, so init this new listener individually
-			if( isInitialized )
-				initializeListener(listener);
+	public void addListener(UnitedServerListener listener) {
+		boolean requiresInit = false;
+		synchronized(this) {

```

Comment:
Changing all calls from synchronized(this) to  synchronized(list)  would have no actual effect and would just be a cosmetic change.  My personal opinion is that synchronizing on (this) makes it more clear that there's only 1 lock for the file. When I see code synchronizing on specific objects, I suspect other parts of the code is synchronizing on other objects.  So this may be more of a style choice, but I prefer synchronizing on (this) when a class has only 1 real lock.